python/ray/autoscaler/_private/vsphere/config.py

```diff
@@ -49,6 +49,46 @@ def bootstrap_vsphere(config):
     return config
 
 
+def validate_frozen_vm_configs(conf: dict):
+    """
+    valid frozen VM configs are:
+    1. ``ray up`` on a frozen VM to be deployed from an OVF template:
+    frozen_vm:
+        name: single-frozen-vm
+        library_item: frozen-vm-template
+        cluster: vsanCluster
+        datastore: vsanDatastore
+
+    2. ``ray up`` on an existing frozen VM:
+        frozen_vm:
+            name: existing-single-frozen-vm
+
+    3. ``ray up`` on a resource pool of frozen VMs to be deployed from an OVF template:
+        frozen_vm:
+            name: frozen-vm-prefix
+            library_item: frozen-vm-template
+            resource_pool: frozen-vm-resource-pool
+            datastore: vsanDatastore
+
+    4. ``ray up`` on an existing resource pool of frozen VMs:
+        frozen_vm:
+            resource_pool: frozen-vm-resource-pool
+    This function will throw an Exception if the config doesn't lie in above examples
+    """
+    # This means deploy from OVF
+    if "library_item" in conf:
+        # Deploy to which datastore must be given
+        assert conf.get("datastore")
+        # Either give a cluster, or a resource_pool. cluster means deploy one frozen VM
+        # resource_pool means deploy a set of frozen VMs
+        assert conf.get("cluster") or conf.get("resource_pool")
+        # name must exist when deploy from OVF
+        assert conf.get("name")
+    else:
+        # If frozen VM(s) exist(s), then just check if name or resource pool presents
+        assert "name" in conf or "resource_pool" in conf
```

Comment:
I think `assert` should be replaced by `ValueError` with a descriptive error message, otherwise users will complain.

Reply:
You can use `with pytest.raises(ValueError, match=...)` for the unit test. https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Reply:
Can be done in a followup PR